.devcontainer/Dockerfile

```diff
@@ -0,0 +1,23 @@
+
+# start from Ubuntu 16.04
+FROM ubuntu:xenial
+
+# install python 3.6, pip, venv and git
+RUN apt-get update
+RUN apt-get -y install software-properties-common
+RUN add-apt-repository ppa:deadsnakes/ppa
+RUN apt-get update
+RUN apt-get -y install python3.6 python3-pip python3-venv git
+
+# clone azure cli into /workspaces/azure-cli
+RUN mkdir /workspaces; cd /workspaces; git clone https://github.com/Azure/azure-cli.git
+
+# create environment in /env and set up 
+RUN python3 -m venv env
```

Comment:
Given that we're running in a container, I think that the environment is already as isolated as it needs to be. Not using a virtual environment here may simplify life a little.